### PR TITLE
Simplify prerequisite for lite development

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -18,31 +18,42 @@ git submodule foreach git pull origin master
 
 ## Prerequisites
 
+### Basic Development
+
+For development that only involves minor bug fix and feature addition, need to:
```
Comment (beets) [inline docs/developer_guide.md:23]:
```suggestion
For development that only involves minor bug fixes or feature additions, the following steps are required:
```

File: docs/developer_guide.md [hunk 2]
```diff
@@ -18,31 +18,42 @@ git submodule foreach git pull origin master
...
+
+- Initialize the mixer submodule
+
+  ```bash
+  git submodule update --init --recursive
+  ```
+
+### Place Search
+
+Development that involves place search need to:
```
Comment (beets) [inline docs/developer_guide.md:35]:
```suggestion
Development that involves place search needs the following additional requirements:
```

File: docs/developer_guide.md [hunk 3]
```diff
@@ -18,31 +18,42 @@ git submodule foreach git pull origin master
...
   gcloud auth application-default login
   ```
 
-- Initialize the mixer submodule
+### Develop with Kubernetes
 
-  ```bash
-  git submodule update --init --recursive
-  ```
+This is only needed if developing in local Kubernetes cluster.
```
Comment (beets) [inline docs/developer_guide.md:49]:
```suggestion
This is only needed for development with a local Kubernetes cluster (e.g. for changes dependent on mixer changes).
```

**Review by beets — APPROVED** [review]
thx for the updated docs!

**Comment by shifucun** [comment]
Thanks for the suggestions!

**Pull request merged**